t/17-basetest.t

```diff
@@ -521,4 +521,20 @@ subtest rollback_activated_consoles => sub {
     is_deeply(\@selected_consoles, [{cmd => 'backend_select_console', testapi_console => 'last_milestone_console'}], 'last milestone console selected') or diag explain \@selected_consoles;
 };
 
+subtest search_for_expected_serial_failures => sub {
+    $bmwqemu::vars{BACKEND} = 'qemu';
+    my $basetest_class = 'basetest';
+    Test::MockModule->new("autotest")->noop('set_current_test');
```

Comment:
This like is not effective. You need to assign `Test::MockModule->new("autotest")` to a variable because if the object's lifetime ends the mocking will be undone.